qa/tasks/salt_manager.py

```diff
@@ -78,13 +78,22 @@ def __cat_file_remote(self, remote, filename=None):
                 ).format(filename, remote.name))
 
     def __ping(self, ping_cmd, expected):
-        with safe_while(sleep=15, tries=50,
-                        action=ping_cmd) as proceed:
-            while proceed():
-                responded = len(re.findall('  True', self.master_remote.sh(ping_cmd)))
-                log.info("{} of {} minions responded".format(responded, expected))
-                if (expected == responded):
-                    return None
+        try:
+            def instances_of_str(search_str, output):
```

Comment:
I think there is no advantage of using this shortcut.

Reply:
it is even longer than len/re combination:
```
>>> len('instances_of_str()')
18
>>> len('len(re.findall())')
17
```

Reply:
The advantage is code readability.

Reply:
In this particular case it even worse readability as for me.